src/celeritas/neutron/model/detail/NuclearZoneBuilder.hh

```diff
@@ -0,0 +1,410 @@
+        real_type par_a = (options_.use_two_params ? 1.16 : 1.2);
+        real_type par_b = (options_.use_two_params ? -1.3456 : 0);
+        nuclear_radius = options_.radius_scale
+                         * (par_a * cbrt_a + par_b / cbrt_a);
+    }
+    else
+    {
+        nuclear_radius = options_.radius_small
+                         * (amass == 4 ? options_.radius_alpha : 1);
+    }
+
+    return nuclear_radius;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Integrate the Woods-Saxon potential, \f$ V(r) \f$,
+ *
+ * \f[
+     V(r) = frac{V_{o}}{1 + e^{\frac{r - R}{a}}}
+   \f]
+ * numerically over the volume from \f$ r_{min} \f$ to \f$ r_{rmax} \f$, where
+ * \f$ V_{o}, R, a\f$ are the potential well depth, nuclear radius, and
+ * surface thickness (skin depth), respectively.
+ */
+real_type
+NuclearZoneBuilder::integrate_woods_saxon(real_type rmin,
+                                          real_type rmax,
+                                          real_type nuclear_radius) const
+{
+    real_type skin_ratio = nuclear_radius / skin_depth_;
+
+    real_type ws_shift = 2 * skin_ratio;
+    real_type delta_r = rmax - rmin;
+
+    // Average value of \f$ r^{2} V(r) dr \f$ at boundaries
+    real_type integral = half() * delta_r
+                         * (rmin * (rmin + ws_shift) / (1 + std::exp(rmin))
+                            + rmax * (rmax + ws_shift) / (1 + std::exp(rmax)));
+
+    real_type result = integrate_potential(rmin, delta_r, integral, ws_shift);
+
+    return ipow<3>(skin_depth_)
+           * (result
+              + ipow<2>(skin_ratio)
+                    * std::log((1 + std::exp(-rmin)) / (1 + std::exp(-rmax))));
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Integrate the Gaussian potential function from \f$ r_1 \f$ to \f$ r_2 \f$.
+ */
+real_type NuclearZoneBuilder::integrate_gaussian(real_type rmin,
+                                                 real_type rmax,
+                                                 real_type gauss_radius) const
+{
+    real_type delta_r = rmax - rmin;
+    real_type rmin_sq = ipow<2>(rmin);
+    real_type rmax_sq = ipow<2>(rmax);
+
+    real_type integral
+        = this->half() * delta_r
+          * (rmin_sq * std::exp(-rmin_sq) + rmax_sq * std::exp(-rmax_sq));
+
+    real_type result = integrate_potential(false, rmin, delta_r, integral);
+
+    return ipow<3>(gauss_radius) * result;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Integrate a nuclear potential numerically by adaptive quadrature.
+ */
+real_type NuclearZoneBuilder::integrate_potential(real_type rmin,
+                                                  real_type delta_r,
+                                                  real_type integral,
+                                                  real_type ws_shift) const
+{
+    constexpr int max_trials = 1e+3;
```

Comment:
@whokion This is the source of the windows error:
```
D:\a\celeritas\celeritas\src\celeritas\neutron\model\detail/NuclearZoneBuilder.hh(368): error C2220: the following warning is treated as an error
2024-06-12T13:55:56.9194698Z D:\a\celeritas\celeritas\src\celeritas\neutron\model\detail/NuclearZoneBuilder.hh(368): warning C4244: 'initializing': conversion from 'double' to 'const int', possible loss of data
```

Write it as `1000`. (In theory you could specify a real number that's out of range of `int` and then loses accuracy.)

Reply:
Thanks!